tw-tasks-executor/src/main/java/com/transferwise/tasks/helpers/kafka/messagetotask/KafkaMessageHandlerFactory.java

```diff
@@ -0,0 +1,94 @@
+package com.transferwise.tasks.helpers.kafka.messagetotask;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.transferwise.tasks.ITasksService;
+import com.transferwise.tasks.utils.TriConsumer;
+import lombok.RequiredArgsConstructor;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.springframework.stereotype.Component;
+
+/** Factory for ready to use {@link IKafkaMessageHandler kafka message handlers}. */
+@Component
+@RequiredArgsConstructor
+public class KafkaMessageHandlerFactory {
+
+    private final ITasksService tasksService;
+    private final ObjectMapper objectMapper;
+
+    /**
+     * Creates default handler, the behaviour of the handler is the same to {@link KafkaMessageToTaskConverter}.
+     * When the handler fails to deserialize a message it will infinitely attempt to deserialize the same message
+     * over and over again, the processing of the other messages in the topic will be stuck.
+     */
+    public <T> IKafkaMessageHandler<String> createDefaultHandler(
+        Class<T> dataObjClass,
+        TriConsumer<T, ConsumerRecord<String, String>, ITasksService.AddTaskRequest> consumer,
+        IKafkaMessageHandler.Topic... topics
+    ) {
+        return createResilientHandler(
+            new InfiniteRetryCorruptedMessageRecoveryStrategy(),
+            dataObjClass,
+            consumer,
+            topics
+        );
+    }
+
+    /**
+     * Creates a resilient handler, every corrupted message will wrapped and
+     * processed separately as described in {@link CorruptedMessageRecoveryStrategy}.
+     *
+     * @see ResilientKafkaMessageHandler
+     */
+    public <T> IKafkaMessageHandler<String> createDefaultResilientHandler(
+        Class<T> dataObjClass,
+        TriConsumer<T, ConsumerRecord<String, String>, ITasksService.AddTaskRequest> consumer,
+        IKafkaMessageHandler.Topic... topics
+    ) {
+        return createResilientHandler(
+            new CreateTaskForCorruptedMessageRecoveryStrategy(tasksService),
+            dataObjClass,
+            consumer,
+            topics
+        );
+    }
+
+    /**
+     * Creates a resilient handler, every corrupted message will wrapped and
+     * processed separately as described in {@link CorruptedMessageRecoveryStrategy}.
+     *
+     * @see ResilientKafkaMessageHandler
+     */
+    public <T> IKafkaMessageHandler<String> createDefaultResilientHandler(
+        Class<T> dataObjClass,
+        TriConsumer<T, ConsumerRecord<String, String>, ITasksService.AddTaskRequest> consumer,
+        String topicAddress
```

Comment:
why this method only accept one topic, when other accept many?

Reply:
This is for convenience as most of the handlers we currently have bound to a single topic which is specified as string